src/app/settings.js

```diff
@@ -280,20 +280,23 @@ var AdvSettings = {
         var _url = url.parse(endpoint.url);
         win.debug('Checking %s endpoint', _url.hostname);
 
+        function tryNextEndpoint() {
+            if (endpoint.index < endpoint.proxies.length - 1) {
+                endpoint.index++;
+                AdvSettings.checkApiEndpoint(endpoint, defer);
+            } else {
+                endpoint.index = 0;
+                endpoint.ssl = undefined;
+                _.extend(endpoint, endpoint.proxies[endpoint.index]);
+                defer.resolve();
+            }
+        }
+
         if (endpoint.ssl === false) {
-            var timeoutWrapper = function (req) {
-                return function () {
-                    win.warn('[%s] Endpoint timed out',
-                        _url.hostname);
-                    req.abort();
-                    tryNextEndpoint();
-                };
-            };
             var request = http.get({
                 hostname: _url.hostname
             }, function (res) {
                 res.once('data', function (body) {
-                    clearTimeout(timeout);
                     res.removeAllListeners('error');
```

Comment:
That's why I said "a great part". I should have mentioned that change, though.

Instead of using a `setTimeout` outside the request to then abort it and retry again; it uses a `setTimeout` from the request object, so that you can get rid of the hassle of `clearTimeout` and variables not defined, etc.

Why it looks wrong ?